src/main/java/io/cdap/e2e/utils/WaitHelper.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.e2e.utils;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.JavascriptExecutor;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Wait helper
+ */
+public class WaitHelper {
+  private static final Logger logger = LoggerFactory.getLogger(WaitHelper.class);
+  private static final JavascriptExecutor javascriptExecutor = (JavascriptExecutor) SeleniumDriver.getDriver();
+
+  public static void waitForPageToLoad() {
+    while (true) {
+      logger.info("Waiting for page to load...");
+      boolean result = (boolean) javascriptExecutor.executeScript("return document.readyState == 'complete'");
+      if (result) {
+        return;
+      }
+    }
+  }
+
+  public static void waitForPageToLoad(long staticWaitInSeconds) {
+    waitForPageToLoad();
+
+    try {
+      logger.info("Additional static wait for page to load and stabilize: " + staticWaitInSeconds);
+      Thread.sleep(staticWaitInSeconds);
```

Comment:
throw an exception in waitForPageToLoad() if it doesn't load within the timeout. In general, avoid waiting forever as it causes tests to hang, makes it hard to debug - throw an exception if the condition you're waiting for is not satisfied within some reasonable timeout.

Reply:
Added timeout for the page load condition.